xbmc/ApplicationPlayer.cpp

```diff
@@ -62,29 +62,31 @@ void CApplicationPlayer::CloseFile(bool reopen)
   }
 }
 
+bool CApplicationPlayer::CanClosePlayerGapless(std::string &playername, CFileItem& item)
+{
+  std::shared_ptr<IPlayer> player = GetInternal();
+  if (!player)
+    return false;
+
+  bool gaplessSupported = player->m_type == "music" || (player->m_type == "video" && !item.IsDiscImage() && !item.IsDVDFile());
+  gaplessSupported = gaplessSupported && (playername == player->m_name);
+  return gaplessSupported;
```

Comment:
so if a player supports gapless, we can close it? why?

Reply:
The gapless logic was there all along. This is to close the player without actually closing it, i.e. open a new file. It sends a message on the VP queue to pick up the new file and start it.